src/test/java/org/jqassistant/plugin/jee/transaction/constraint/TransactionalMethodMustNotBeInvokedFromSameSubclassIT.java

```diff
@@ -0,0 +1,294 @@
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.jakarta.JakartaCallingSubClassOfSimpleTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.jakarta.JakartaOverridingSubClassOfSimpleClassWithTransactionalMethod;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.jakarta.JakartaOverridingSubClassOfSimpleNonTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.jakarta.JakartaOverridingSubClassOfSimpleTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.jakarta.JakartaSimpleClassWithTransactionalMethod;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.jakarta.JakartaSimpleNonTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.jakarta.JakartaSimpleTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxCallingSubClassOfSimpleClassWithTransactionalMethod;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxCallingSubClassOfSimpleNonTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxCallingSubClassOfSimpleTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxOverridingSubClassOfSimpleClassWithTransactionalMethod;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxOverridingSubClassOfSimpleNonTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxOverridingSubClassOfSimpleTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxSimpleClassWithTransactionalMethod;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxSimpleNonTransactionalClass;
+import org.jqassistant.plugin.jee.transaction.set.inheritance.simple.javax.JavaxSimpleTransactionalClass;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.buschmais.jqassistant.plugin.java.test.assertj.MethodDescriptorCondition.methodDescriptor;
+import static com.buschmais.jqassistant.plugin.java.test.assertj.TypeDescriptorCondition.typeDescriptor;
+import static org.assertj.core.api.Assertions.assertThat;
+import static com.buschmais.jqassistant.core.report.api.model.Result.Status.FAILURE;
+import static org.assertj.core.api.InstanceOfAssertFactories.type;
+import static org.jqassistant.plugin.jee.SimpleMethodDescriptorCondition.simpleMethodDescriptor;
+
+class TransactionalMethodMustNotBeInvokedFromSameSubclassIT extends AbstractJavaPluginIT {
```

Comment:
Consider changing the name to "TransactionalMethodMustNotBeInvokedFromSameClassOrSubclassIT"